app/src/main/java/com/orange/ouds/app/ui/TopBar.kt

```diff
@@ -28,6 +39,44 @@ fun TopBar(
 ) {
     TopAppBar(
         title = { Text(text = topBarState.title, modifier = Modifier.semantics { traversalIndex = -1f }) },
-        modifier = Modifier.semantics { isTraversalGroup = true }
+        modifier = Modifier.semantics { isTraversalGroup = true },
+        actions = {
+            topBarState.actions.forEach { topBarAction ->
+                topBarAction()
+            }
+        }
     )
+}
+
+enum class TopBarAction {
+    ChangeTheme, ChangeMode;
+
+    @Composable
+    fun TopBarIconButton(onActionClick: (TopBarAction) -> Unit) = when (this) {
+        ChangeTheme -> ChangeThemeAction(onActionClick)
+        ChangeMode -> ChangeModeAction(onActionClick)
+    }
+}
+
+@Composable
+fun getDefaultActions(onActionClick: (TopBarAction) -> Unit): List<@Composable () -> Unit> =
+    TopBarDefaultActions.map { { it.TopBarIconButton(onActionClick = onActionClick) } }
+
+@Composable
+private fun ChangeThemeAction(onClick: (TopBarAction) -> Unit) {
+    IconButton(onClick = { onClick(TopBarAction.ChangeTheme) }) {
+        Icon(painter = painterResource(id = R.drawable.ic_palette), contentDescription = stringResource(id = R.string.app_topBar_changeTheme_button_a11y))
+    }
+}
+
+@Composable
+private fun ChangeModeAction(onClick: (TopBarAction) -> Unit) {
+    val configuration = LocalConfiguration.current
+
+    val painterRes = if (configuration.isDarkModeEnabled) R.drawable.ic_ui_light_mode else R.drawable.ic_ui_dark_mode
+    val iconDesc = if (configuration.isDarkModeEnabled) R.string.app_topBar_changeModeToLight_button_a11y else R.string.app_topBar_changeModeToDark_button_a11y
+
+    IconButton(onClick = { onClick(TopBarAction.ChangeMode) }) {
```

Comment:
`TopBarAction` is grayed out by Android Studio because it's already imported. Same comment line 79.